File: app/content/Article.scala
package content

import com.gu.openplatform.contentapi.model.{ItemResponse, Content => ApiContent}
import frontend.common._
import conf._

case class Article(private val content: ApiContent, private val related: Seq[Trail] = Nil)
  extends Content(content, related)  {

  lazy val body: String = content.safeFields("body")

  override lazy val metaData = super.metaData + ("content-type" -> "Article")
}

object Article extends Logging {

  def byId(path: String): Option[Article] = suppressApi404 {
    log.info("Fetching article: " + path)
    val response: ItemResponse = ContentApi.item
      .showTags("all")
      .showFields("all")
      .showMedia("all")
      .showRelated(true)
      .itemId(path)
      .response
    val related = response.relatedContent map  { Trail(_) }

    response.content.filter { _.isArticle } map { Article(_, related) }
  }
}

File: tools/perf-test/models/article.rb
class Article 
    
    include ContentApi

    def initialize(path)
        super
        @params.show_fields = 'all'
        @params.show_inline_elements = 'picture'
        @params.show_tags = 'all'
        @params.show_media = 'all'
        @params.show_story_package = 'true'
        @params.tag = 'type/gallery,type/article,type/video'
    end

end
